spyder_kernels/console/kernel.py

```diff
@@ -126,30 +124,77 @@ def frontend_call(self, blocking=False, broadcast=True,
             callback=callback,
             timeout=timeout)
 
-    def enable_faulthandler(self, fn):
+    def enable_faulthandler(self):
         """
         Open a file to save the faulthandling and identifiers for
         internal threads.
         """
-        self.disable_faulthandler()
-        f = open(fn, 'w')
-        self.faulthandler_handle = f
-        f.write("Main thread id:\n")
-        f.write(hex(threading.main_thread().ident))
-        f.write('\nSystem threads ids:\n')
-        f.write(" ".join([hex(thread.ident) for thread in threading.enumerate()
-                          if thread is not threading.main_thread()]))
-        f.write('\n')
-        faulthandler.enable(f)
-
-    def disable_faulthandler(self):
-        """
-        Cancel the faulthandling, close the file handle and remove the file.
-        """
-        if self.faulthandler_handle:
-            faulthandler.disable()
-            self.faulthandler_handle.close()
-            self.faulthandler_handle = None
+        self.faulthandler_handle = tempfile.NamedTemporaryFile(
+            'wt', suffix='.fault')
```

Comment:
I am not sure how to do that? On mac at least this create a file like: `'/var/folders/xx/mbs7xxcs2397g51c4kbg23p00000gp/T/tmps5mzxk5n.fault'`

Reply:
It doesn't look like this can be easily done with `tempfile.NamedTemporaryFile`. We could add linux specific code but that would add complexity to the code and possible bugs down the line.

Reply:
Maybe I should expand:
 - What I need here is a temporary file that would be destroyed as soon as the kernel stops. The only way for the file to not disappear should be for the kernel to crash.
 - `tempfile.NamedTemporaryFile` Fits this description perfectly and should work as advertised for all OS, but doesn't allow the automatic creation of folders.
 - `tempfile.TemporaryDirectory` would not work because it would create a directory for each kernel.
 - The solution would be to create for linux `tmp/spyder-kernels`but that means creating a directory that might not be deletable easily. Also this solution would be os-specific.

Reply:
> The solution would be to create for linux /tmp/spyder-kernelsbut that means creating a directory that might not be deletable easily. Also this solution would be os-specific.

I understand, but I'm really against polluting `/tmp` on Linux (as many apps do these days). So please implement a specific solution for that OS.

To do that, I suggest to add as a new dependency the `pyxdg` package, use

https://pyxdg.readthedocs.io/en/latest/basedirectory.html#xdg.BaseDirectory.xdg_data_home

to get the data directory per user, create or check that `spyder` exists under it and use it to save these files. Only if that fails, you should resort to your current approach of saving them directly in `/tmp`.

Reply:
Alternatively I could create `~/.local/share` or `/tmp/spyder` and save the files in this folder?

Reply:
> Alternatively I could create ~/.local/share

That's the default XDG directory but it can be changed by setting an env var.

> /tmp/spyder and save the files in this folder?

That wouldn't work on a multi-user system because different users are not able to write to the same directory.

So, I think it's simpler and better to use `pyxdg`, which is a very lightweight dependency, and avoid to handle that ourselves.

Reply:
Done but I am unsure if the dependency is correctly added for conda

Reply:
Thanks, I'll leave a comment so you add it to the right file.